### PR TITLE
[airframe-rx-html] Make event handlers type-safe

### LINKED ISSUE #1599: airframe-rx-html: Type-safe event handler
onclick event handler binding should be type safe. 

Current:
```scala
div(onclick -> { e: MouseEvent => ... }) // No function type check
```

Idea:
```scala
div(onClick -> {e: MouseEvent => ...}) // Only Function1[MouseEvent, Unit] can be used
```

### PR DESCRIPTION
**Pull request by exoego**
Closes #1599 

Handlers are tested in https://github.com/wvlet/airframe/blob/master/airframe-rx-html/.js/src/test/scala/wvlet/airframe/http/rx/html/HtmlRenderingTest.scala

### CODE REVIEW

File: airframe-rx-html/src/main/scala/wvlet/airframe/rx/html/package.scala [hunk 1]
```diff
@@ -59,6 +59,14 @@ package object html {
     def noValue: HtmlNode                             = HtmlAttribute(name, true, namespace)
   }
 
+  class HtmlEventHandlerOf[E](name: String, namespace: Namespace = Namespace.xhtml) {
+    def apply[U](v: E => U): HtmlNode  = HtmlAttribute(name, v, namespace)
+    def ->[U](v: E => U): HtmlNode     = apply(v)
+    def apply[U](v: () => U): HtmlNode = HtmlAttribute(name, v, namespace)
+    def ->[U](v: () => U): HtmlNode    = apply(v)
```
Comment (exoego) [inline airframe-rx-html/src/main/scala/wvlet/airframe/rx/html/package.scala:66]: - `E => U` is for `e: MouseEvent => ...` 
- `() => U` is for ` () =>. ...` which ignores event

File: airframe-rx-html/src/main/scala/wvlet/airframe/rx/html/Attrs.scala [hunk 2]
```diff
@@ -527,98 +527,98 @@ trait MouseEventAttrs {
...
-  lazy val onmouseup = attr("onmouseup")
+  lazy val onmouseup = handler[compat.MouseEvent]("onmouseup")
 
   /**
     * Specifies the function to be called when the window is scrolled.
     *
     * MDN
     */
-  lazy val onscroll = attr("onscroll")
+  lazy val onscroll = handler[compat.UIEvent]("onscroll")
```
Comment (exoego) [inline airframe-rx-html/src/main/scala/wvlet/airframe/rx/html/Attrs.scala:616]: Note) onscroll receives `UIEvent`, not `MouseEvent`
https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onscroll

File: airframe-rx-html/src/main/scala/wvlet/airframe/rx/html/Attrs.scala [hunk 3]
```diff
@@ -527,98 +527,98 @@ trait MouseEventAttrs {
...
 
   /**
     * Fires when the mouse wheel rolls up or down over an element
     */
-  lazy val onwheel = attr("onwheel")
+  lazy val onwheel = handler[compat.WheelEvent]("onwheel")
```
Comment (exoego) [inline airframe-rx-html/src/main/scala/wvlet/airframe/rx/html/Attrs.scala:621]: Note)
Note) onscroll receives `WheelEvent`, not `MouseEvent`
https://developer.mozilla.org/en-US/docs/Web/API/Element/wheel_event
https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onwheel

**Comment by xerial** [comment]
Awesome! Thank you for the improvement. I wanted to have this type safety for a while. airframe-rx will be more comfortable to use in modern editors like IntelliJ.

**Pull request merged**